src/index.ts

```diff
@@ -337,21 +337,39 @@ export class Completion {
     previousArgs: string[],
     toComplete: string
   ) {
-    const commandParts = [...previousArgs];
+    const commandParts = [...previousArgs].filter(arg => arg !== '');
+
+    // When we have no real command parts (just empty or showing root commands)
+    const isRootCompletion = commandParts.length === 0;
 
     for (const [k, command] of this.commands) {
       if (k === '') continue;
+
       const parts = k.split(' ');
-      let match = true;
 
+      // For root completion, just match the first part of each command
+      if (isRootCompletion) {
```

Comment:
I do not think this is necessary or it should be that explicit! The while loop below that handles subcommands should already handle this.